### PR TITLE
Limit V2 segment file row count (#1647)

### PR DESCRIPTION
**Pull request by liutang123**
Because many bitmap implementations usually only support 32 bits integer,
so use INT_MAX to limit the row count of a segment file.

### CODE REVIEW

**Comment by liutang123** [comment]
@gaodayue

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -84,6 +84,9 @@ OLAPStatus BetaRowsetWriter::_add_row(const RowType& row) {
     if (PREDICT_FALSE(_segment_writer == nullptr)) {
         RETURN_NOT_OK(_create_segment_writer());
     }
+    if (PREDICT_FALSE(_segment_writer->reach_capacity())) {
```
Comment (chaoyli) [inline be/src/olap/rowset/beta_rowset_writer.cpp]: You can check with _max_segment_size together.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_v2/segment_writer.h [hunk 2]
```diff
@@ -60,6 +60,8 @@ class SegmentWriter {
 
     uint64_t estimate_segment_size();
 
+    bool reach_capacity();
```
Comment (gaodayue) [inline be/src/olap/rowset/segment_v2/segment_writer.h]: I prefer to have a function named `num_rows_written()` and move the logic of when to flush segment to BetaRowsetWriter.

File: be/src/olap/rowset/segment_v2/segment_writer.h [hunk 3]
```diff
@@ -82,6 +84,7 @@ class SegmentWriter {
     std::unique_ptr<WritableFile> _output_file;
     std::vector<ColumnWriter*> _column_writers;
     uint64_t _row_count = 0;
+    const int32_t MAX_ROW_COUNT = INT_MAX;
```
Comment (gaodayue) [inline be/src/olap/rowset/segment_v2/segment_writer.h]: INT_MAX is not the same on different platforms, better to use INT32_MAX.

File: be/src/olap/rowset/beta_rowset_writer.cpp (hunk 1 above)
Comment (gaodayue) [reply be/src/olap/rowset/beta_rowset_writer.cpp]: +1

**Review by gaodayue — CHANGES_REQUESTED** [review]

**Comment by gaodayue** [comment]
In addition
- `SegmentFooterPB.num_rows` should be changed from uint64 to uint32
- `segment_v2::SegmentWriter::_row_count` should be changed from uint64_t to uint32_t

**Review by gaodayue — APPROVED** [review]
LGTM

**Review by chaoyli — APPROVED** [review]

**Comment by chaoyli** [comment]
LGTM, I will merge it after code checks.

File: be/src/olap/rowset/segment_v2/segment_writer.h [hunk 4]
```diff
@@ -60,6 +60,8 @@ class SegmentWriter {
 
     uint64_t estimate_segment_size();
 
+    uint32_t num_rows_written() { return _row_count; }
```
Comment (imay) [inline be/src/olap/rowset/segment_v2/segment_writer.h]:
```suggestion
    uint32_t num_rows_written() const { return _row_count; }
```

**Pull request merged**